sw/airborne/modules/nav/nav_skid_landing.c

```diff
@@ -0,0 +1,199 @@
+/*
+ *
+ * Copyright (C) 2016, Michal Podhradsky, Thomas Fisher
+ *
+ * AggieAir, Utah State University
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ */
+
+/**
+ * @file modules/nav/nav_skid_landing.c
+ * @brief Landing on skidpads
+ * See video of the landing: https://www.youtube.com/watch?v=aYrB7s3oeX4
+ * Standard landing procedure:
+ * 1) circle down passing AF waypoint (from left or right)
+ * 2) once low enough follow line to TD waypoint
+ * 3) once low enough flare
+ *
+ * Use this in your airfame config file:
+ *   <section name="LANDING" prefix="SKID_LANDING_">
+ *     <define name="AF_HEIGHT" value="50" unit="m"/>
+ *     <define name="FINAL_HEIGHT" value="50" unit="m"/>
+ *     <define name="FINAL_STAGE_TIME" value="10" unit="s"/>
+ *   </section>
+ */
+
+#include "generated/airframe.h"
+#include "state.h"
+#include "modules/nav/nav_skid_landing.h"
+#include "firmwares/fixedwing/autopilot.h"
+#include "firmwares/fixedwing/nav.h"
+#include "firmwares/fixedwing/stabilization/stabilization_attitude.h"
+
+enum LandingStatus
+{
+  CircleDown, LandingWait, Final, Approach
+};
+static enum LandingStatus skid_landing_status;
+static uint8_t aw_waypoint;
+static uint8_t td_waypoint;
+static float land_radius;
+static struct FloatVect2 land_circle;
+static float land_app_alt;
+static float land_circle_quadrant;
+static float approach_quadrant;
+static float final_land_altitude;
+static uint8_t final_land_count;
+
+#ifndef SKID_LANDING_AF_HEIGHT //> AF height [m]
+#define SKID_LANDING_AF_HEIGHT 50
+#endif
+#ifndef SKID_LANDING_FINAL_HEIGHT //> final height [m]
+#define SKID_LANDING_FINAL_HEIGHT 5
+#endif
+#ifndef SKID_LANDING_FINAL_STAGE_TIME //> final stage time [s]
+#define SKID_LANDING_FINAL_STAGE_TIME 5
+#endif
+
+static inline float distance_equation(struct FloatVect2 p1,struct FloatVect2 p2)
+{
+  return sqrt((p1.x-p2.x)*(p1.x-p2.x)+(p1.y-p2.y)*(p1.y-p2.y));
```

Comment:
for float values use `sqrtf` or even directly something like
```
struct FloatVect2 diff;
VECT2_DIFF(diff, p1, p2);
return float_vect2_norm(diff);
```
or even add a distance function in math?